server/src/main/java/org/candlepin/model/Product.java

```diff
@@ -1124,53 +1123,62 @@ public boolean equals(Object obj) {
 
     @Override
     public int hashCode() {
+        int hash = 7;
+
+        hash = 17 * hash + (this.id != null ? this.id.hashCode() : 0);
+
+        return hash;
+    }
+
+    /**
+     * Calculates and returns a version hash for this entity. This method operates much like the
+     * hashCode method, except that it is more accurate and should have fewer collisions.
+     *
+     * @return
+     *  a version hash for this entity
+     */
+    public int getEntityVersion() {
         // This must always be a subset of equals
         HashCodeBuilder builder = new HashCodeBuilder(37, 7)
```

Comment:
We only use the .hashCode method on objects which have explicitly defined an implementation which is consistent between instances. We also have tests which verify this.